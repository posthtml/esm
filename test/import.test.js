const posthtml = require('./helpers')
const { imports } = require('../lib')

describe('Import', () => {
  describe('Defaults', () => {
    test('src', () => {
      const plugins = [ imports() ]
      const options = {}

      return posthtml('import.html', plugins, options)
        .then(({ html, messages }) => {
          expect(html).toMatchSnapshot()
          expect(messages).toMatchSnapshot()
        })
    })
  })

  describe('Filter', () => {
    test('src', () => {
      const plugins = [
        imports({
          import: /filter/
        })
      ]
      const options = {}

      return posthtml('import.html', plugins, options)
        .then(({ html, messages }) => {
          expect(html).toMatchSnapshot()
          expect(messages).toMatchSnapshot()
        })
    })
  })
})
