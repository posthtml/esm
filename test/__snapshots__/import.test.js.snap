// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Import Defaults src 1`] = `
"\${HTML__IMPORT__0}
\${HTML__IMPORT__1}
"
`;

exports[`Import Defaults src 2`] = `
Array [
  Object {
    "import": [Function],
    "name": "HTML__IMPORT__0",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./import.html",
  },
  Object {
    "import": [Function],
    "name": "HTML__IMPORT__1",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./filter/import.html",
  },
]
`;

exports[`Import Filter src 1`] = `
"\${HTML__IMPORT__0}

"
`;

exports[`Import Filter src 2`] = `
Array [
  Object {
    "import": [Function],
    "name": "HTML__IMPORT__0",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./import.html",
  },
  Object {
    "import": [Function],
    "name": "HTML__IMPORT__1",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./filter/import.html",
  },
  Object {
    "import": [Function],
    "name": "HTML__IMPORT__0",
    "plugin": "@posthtml/esm",
    "type": "import",
    "url": "./import.html",
  },
]
`;
